packages/paste-core/components/combobox/src/ComboboxItems.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import type {VirtualItem} from 'react-virtual/types';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {ComboboxListboxOption} from './styles/ComboboxListboxOption';
+import {ComboboxListboxGroup} from './styles/ComboboxListboxGroup';
+import {getIndexedItems, getGroupedItems} from './helpers';
+import type {ComboboxItemsProps} from './types';
+
+const ComboboxItems: React.FC<ComboboxItemsProps> = ({
+  items,
+  getItemProps,
+  highlightedIndex,
+  optionTemplate,
+  groupLabelTemplate,
+  groupItemsBy,
+  rowVirtualizer,
+}) => {
+  const UIDSeed = useUIDSeed();
+
+  // If no groupings, return plain list
+  if (groupItemsBy == null) {
+    // If there's an option template, disable virtualization
+    // because calcing dynamic heights is hard
+    // TODO: virtualize
+    return optionTemplate != null ? (
+      <ComboboxListboxGroup>
+        {items.map((item, index) => (
+          <ComboboxListboxOption
+            {...getItemProps({item, index})}
+            highlighted={highlightedIndex === index}
+            key={UIDSeed(`item-${index}`)}
+            variant="default"
+          >
+            {optionTemplate(item)}
+          </ComboboxListboxOption>
+        ))}
+      </ComboboxListboxGroup>
+    ) : (
+      <ComboboxListboxGroup>
+        <li role="presentation" key="total-size" style={{height: rowVirtualizer!.totalSize}} />
+        {rowVirtualizer!.virtualItems.map((virtualItem: VirtualItem) => {
+          const item = items[virtualItem.index];
+          return (
+            <ComboboxListboxOption
+              // @ts-ignore
+              ref={(element) => virtualItem.measureRef(element)}
+              {...getItemProps({item, index: virtualItem.index})}
+              highlighted={highlightedIndex === virtualItem.index}
+              key={UIDSeed(`item-${virtualItem.index}`)}
```

Comment:
Super nit: should we make `virtualItem.index` a `const` since we're re-using it here?